### PR TITLE
Scout Drone Control

### PR DESCRIPTION
**Pull request by acarrou**
Adds necessary drone control for Scout drones. This includes simulation and a Python demo script that uses the mavsdk c++ script as a subprocess

### CODE REVIEW

**Comment by acarrou** [comment]
@naterpotatoers Please review this.

File: scout/modules/FlightProcess.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        threading.Thread(target=self.read_from_process, args=(self.cpp_process,self.WaypointGen,), daemon=True).start()
+        return self.cpp_process
+
+
+    def read_from_process(self):
+        while True:
+            line = self.cpp_process.stdout.readline().strip()
+            if not line:
+                break
+            print(Fore.YELLOW + "Received from C++:", Fore.BLUE, f"{line}", Style.RESET_ALL)
```
Comment (naterpotatoers) [inline scout/modules/FlightProcess.py:23]:
```suggestion
            print_cpp(...)
```

File: scout/scout_search/scout_search.cpp [hunk 2]
```diff
@@ -123,32 +153,61 @@ void resume_mission(Mission& mission)
...
+        mission.set_return_to_launch_after_mission(true);
         upload_mission(mission, mission_items);
 
-    }else if (cmd == "add_waypoint") {
+    } else if (cmd == "upload_no_rtl") {
+        mission.set_return_to_launch_after_mission(false);
+        upload_mission(mission, mission_items);
+    } else if (cmd == "status") {
+        drone_status(telemetry, action, mission);
+    } else if (cmd == "add_waypoint") {
```
Comment (naterpotatoers) [inline scout/scout_search/scout_search.cpp:194]: why not use a switch statement?

**Review by naterpotatoers — APPROVED** [review]
👍

File: scout/scout_search/scout_search.cpp (hunk 2 above)
Comment (acarrou) [reply scout/scout_search/scout_search.cpp:194]: true it would be more efficient in this case

**Pull request merged**